snapshot/snapshot.go

```diff
@@ -284,6 +291,22 @@ func NewSnapshotter(ctx context.Context, cfg *config.SnapshotterConfig) (snapsho
 		syncRemove = true
 	}
 
+	// Initialize containerd client for lazy parent recovery.
+	// This allows the snapshotter to query containerd for snapshot
+	// metadata when recovering from desynchronization scenarios where
+	// local BoltDB is missing entries.
+	var containerdClient *client.Client
+	containerdAddr := cfg.ContainerdAddress
+	if containerdAddr == "" {
+		containerdAddr = defaultContainerdAddress
+	}
+	containerdClient, err = client.New(containerdAddr)
```

Comment:
What if containerd restarts? We should not expect `containerdClient` to work all the time.